eval.go

```diff
@@ -124,11 +125,16 @@ func (fr *Frame) Position() syntax.Position {
 	if fr.posn.IsValid() {
 		return fr.posn // leaf frame only (the error)
 	}
-	return fr.fn.funcode.Position(fr.callpc) // position of active call
+	if fn, ok := fr.callable.(*Function); ok {
+		return fn.funcode.Position(fr.callpc) // position of active call
+	}
+	return syntax.MakePosition(&builtinFilename, 1, 0)
 }
 
-// Function returns the frame's function, or nil for the top-level of a module.
-func (fr *Frame) Function() *Function { return fr.fn }
+var builtinFilename = "<builtin>"
```

Comment:
Nit: `const` instead of `var`?

Reply:
It must be a variable as we needs its address on line 131. (The reason the variable is global, not local to that function, is to avoid allocating it more than once.)